### PR TITLE
Ensure tests in NUGet.Commandline.Test and MSBuild.Integration.Test don't rely on the repo root configuration.

### LINKED ISSUE #11444: Convert tests using TestDirectory to SimpleTestPathContext Part2
It's related to https://github.com/NuGet/NuGet.Client/pull/4279
We have many tests using `TestDirectory` method which may pollute global packages folder since it doesn't clean up temporary created assets (causes flakiness). In order to onboard `PackageSourceMapping` we need to convert some to `SimpleTestPathContext` which is cleans up temporary created assets.  This will reduce overall flakiness of CI test pipeline.
About 200 tests failing now: https://devdiv.visualstudio.com/DevDiv/_build/results?buildId=5517484&view=results

**Here're about 64 tests from `NuGet.CommandLine.Test.NuGetInstallCommandTest` assembly, there're other tests from this same assembly [I assigned to myself](https://github.com/NuGet/Home/issues/11436). So please focus only those 64 tests, please.**

Msbuild.Integration.Test.MsbuildRestoreTaskTests.MsbuildRestore_NoPackageSourceMappingsection_NoSourceRelatedLogMessage
NuGet.CommandLine.Test.RestoreNetCoreTest.RestoreNetCore_VerifyOrderOfConfigsAsync
NuGet.CommandLine.Test.NuGetInstallCommandTest.InstallCommand_FromPackagesConfigFile_SpecifyingRelativeSolutionDir
NuGet.CommandLine.Test.NuGetInstallCommandTest.InstallCommand_FromPackagesConfigFile_SpecifyingSolutionDir
NuGet.CommandLine.Test.NuGetInstallCommandTest.InstallCommand_RunTwiceWithVersionSpecifiedVerifyExitCode
NuGet.CommandLine.Test.NuGetInstallCommandTest.InstallCommand_DependencyResolution
NuGet.CommandLine.Test.NuGetInstallCommandTest.InstallCommand_FromPackagesConfigFile
NuGet.CommandLine.Test.NuGetInstallCommandTest.InstallCommand_FromPackagesConfigFileWithExcludeVersion
NuGet.CommandLine.Test.NuGetInstallCommandTest.InstallCommand_WithVersionSpecified
NuGet.CommandLine.Test.NuGetInstallCommandTest.InstallCommand_WithPrereleaseVersionSpecified
NuGet.CommandLine.Test.NuGetInstallCommandTest.InstallCommand_FromPackagesConfigFile_VerifyNoopRestoreExitCode
NuGet.CommandLine.Test.NuGetInstallCommandTest.InstallCommand_GetLastestPrereleaseVersion
NuGet.CommandLine.Test.NuGetInstallCommandTest.InstallCommand_GetLastestReleaseVersion
NuGet.CommandLine.Test.NuGetRestoreCommandTest.RestoreCommand_WithSubsequentRestores_PackageSaveModeIsObserved
NuGet.CommandLine.Test.NuGetRestoreCommandTest.RestoreCommand_LongPathPackage
NuGet.CommandLine.Test.NuGetRestoreCommandTest.RestoreCommand_SourceLoggingFileSource
NuGet.CommandLine.Test.NuGetRestoreCommandTest.RestoreCommand_FromSolutionFile_ProjectsInParentDir
NuGet.CommandLine.Test.NuGetRestoreCommandTest.RestoreCommand_LegacySolutionLevelPackages_DuplicatePackageIds
NuGet.CommandLine.Test.NuGetRestoreCommandTest.RestoreCommand_PackageSourceMappingEmptyPackageSource_Succeed
NuGet.CommandLine.Test.NuGetRestoreCommandTest.RestoreCommand_LegacySolutionLevelPackages_SolutionFile
NuGet.CommandLine.Test.NuGetRestoreCommandTest.RestoreCommand_PreservesPackageFilesTimestamps
NuGet.CommandLine.Test.NuGetRestoreCommandTest.RestoreCommand_NoPackageSourceMappingsection_NoSourceRelatedLogMessage
NuGet.CommandLine.Test.NuGetRestoreCommandTest.RestoreCommand_FromSolutionFile_ReportsNothingToDoWithoutError
NuGet.CommandLine.Test.NuGetRestoreCommandTest.RestoreCommand_SourceLoggingFromGlobalPackageCache
NuGet.CommandLine.Test.NuGetRestoreCommandTest.RestoreCommand_FromPackagesConfigFile
NuGet.CommandLine.Test.NuGetRestoreCommandTest.RestoreCommand_LegacySolutionLevelPackages_SolutionDirectory
NuGet.CommandLine.Test.NuGetRestoreCommandTest.RestoreCommand_FromSolutionFileWithMsbuildPath
NuGet.CommandLine.Test.NuGetRestoreCommandTest.RestoreCommand_LegacySolutionLevelPackages_NoArgument
NuGet.CommandLine.Test.NuGetRestoreCommandTest.RestoreCommand_ProjectContainsSolutionDirs
NuGet.CommandLine.Test.NuGetRestoreCommandTest.RestoreCommand_FromProjectJson_RelativeGlobalPackagesFolder
NuGet.CommandLine.Test.NuGetRestoreCommandTest.RestoreCommand_OneSolutionFileInDirectory
NuGet.CommandLine.Test.NuGetRestoreCommandTest.RestoreCommand_FromHttpSource
NuGet.CommandLine.Test.NuGetRestoreCommandTest.RestoreCommand_SourceLoggingFromV3FeedNoGlobalPackageCache
NuGet.CommandLine.Test.NuGetRestoreCommandTest.RestoreCommand_PackageSourceMappingFilter_Succeed
NuGet.CommandLine.Test.NuGetRestoreCommandTest.RestoreCommand_MissingNuspecFileInPackage_FailsWithNU5037
NuGet.CommandLine.Test.NuGetRestoreCommandTest.RestoreCommand_FromPackagesConfigFileWithOptionSolutionDir
NuGet.CommandLine.Test.NuGetRestoreCommandTest.RestoreCommand_FromProjectFile
NuGet.CommandLine.Test.NuGetRestoreCommandTest.RestoreCommand_FromSolutionFile
NuGet.CommandLine.Test.NuGetRestoreCommandTest.RestoreCommand_WithAuthorSignedPackage_Succeeds
NuGet.CommandLine.Test.NuGetRestoreCommandTest.RestoreCommand_ProjectCannotBeLoaded
NuGet.CommandLine.Test.NuGetUpdateCommandTests.UpdateCommand_FromProjectConfig
NuGet.CommandLine.Test.NuGetUpdateCommandTests.UpdateCommand_Success_ProjectFile_References
NuGet.CommandLine.Test.NuGetUpdateCommandTests.UpdateCommand_Success_CustomPackagesFolder_AbsolutePath
NuGet.CommandLine.Test.NuGetUpdateCommandTests.UpdateCommand_Success_Version_Downgrade
NuGet.CommandLine.Test.NuGetUpdateCommandTests.UpdateCommand_Success_Prerelease_With_Id
NuGet.CommandLine.Test.NuGetUpdateCommandTests.UpdateCommand_Success_Update_DeletedFile
NuGet.CommandLine.Test.NuGetUpdateCommandTests.UpdateCommand_NF_Project_Success
NuGet.CommandLine.Test.NuGetUpdateCommandTests.UpdateCommand_Success_Prerelease
NuGet.CommandLine.Test.NuGetUpdateCommandTests.UpdateCommand_Success_ContentFiles
NuGet.CommandLine.Test.NuGetUpdateCommandTests.UpdateCommand_DependencyResolution_Success
NuGet.CommandLine.Test.NuGetUpdateCommandTests.UpdateCommand_Success_PackagesConfig_References
NuGet.CommandLine.Test.NuGetUpdateCommandTests.UpdateCommand_Success_References_MultipleProjects
NuGet.CommandLine.Test.NuGetUpdateCommandTests.UpdateCommand_Success_References
NuGet.CommandLine.Test.NuGetUpdateCommandTests.UpdateCommand_Success_Version_Upgrade
NuGet.CommandLine.Test.NuGetUpdateCommandTests.UpdateCommand_Success_CustomPackagesFolder_RelativePath
NuGet.CommandLine.Test.RestoreRetryTests.RestoreRetry_ProjectJsonRetryOnFailingV3Source
NuGet.CommandLine.Test.RestoreRetryTests.RestoreRetry_PackagesConfigRetryOnFailingV3Source
NuGet.CommandLine.Test.RestoreRetryTests.RestoreRetry_ProjectJsonRetryOnFailingV2Source
NuGet.CommandLine.Test.RestoreRetryTests.RestoreRetry_PackagesConfigRetryOnFailingV2Source
NuGet.CommandLine.Test.RestoreProjectJsonTest.RestoreProjectJson_MinClientVersionFailAsync
NuGet.CommandLine.Test.RestoreProjectJsonTest.RestoreProjectJson_RestoreFiltersToStablePackages
NuGet.CommandLine.Test.RestoreProjectJsonTest.RestoreProjectJson_FloatReleaseLabelTakesStable
NuGet.CommandLine.Test.RestoreProjectJsonTest.RestoreProjectJson_RestoreFromSlnUsesNuGetFolderSettingsAsync
NuGet.CommandLine.Test.RestoreProjectJsonTest.RestoreProjectJson_GenerateTargetsFileFromNuProj

### PR DESCRIPTION
**Pull request by nkolev92**
## Bug

Fixes: https://github.com/nuget/home/issues/11444

Regression? Last working version:

## Description

Self descriptive. 
This change involves a lot of TestDirectory to SimpleTestPathContext changes (which has a nuget.config file that isolates *almost* everything). 

This change unblocks https://github.com/NuGet/NuGet.Client/pull/4279.

Another big change is ensuring the default configs actually clear the package source mapping configuration.

## PR Checklist

- [x] PR has a meaningful title
- [x] PR has a linked issue.
- [x] Described changes

- **Tests**
  - [x] Automated tests added
  - **OR**
  
  - [ ] Test exception
  - **OR**
  - [ ] N/A 

- **Documentation**
  
  - [ ] Documentation PR or issue filled
  - **OR**
  - [x] N/A

### CODE REVIEW

**Review by jeffkl — APPROVED** [review]

**Comment by erdembayar** [comment]
@nkolev92 
I merged this to my [PackageSourceMapping branch](https://github.com/NuGet/NuGet.Client/tree/dev-eryondon-packagesourcemapping-1201) then still got [some errors](https://devdiv.visualstudio.com/DevDiv/_build/results?buildId=5597705&view=ms.vss-test-web.build-test-results-tab). Just wondering if all errors are addressed here. 
For example `RestoreCommand_LongPathPackage` test mentioned in original 64 error is still failing.

**Comment by nkolev92** [comment]
@erdembayar 

I copied only the nuget.config from your PR (it has way too many commits for me to rebase), and when I was running the tests it seemed like they were passing. 

I'll take another look.

**Comment by nkolev92** [comment]
Looking at the code (not having yet run it), it seems like it should stillb e failing. 

Not 100% sure how I missed it. 
Likely the test was relying on own global packages folder which was probably polluted when those tests were run earlier. 

I'll do another pass + check the tests that are still failing on your branch.

**Comment by nkolev92** [comment]
@erdembayar 

I have done another pass. 

Think the tests should be passing now.

**Comment by erdembayar** [comment]
> @erdembayar
> 
> I have done another pass.
> 
> Think the tests should be passing now.

Thank you. I queued new CI job with merged branch, let's see new result.

**Comment by erdembayar** [comment]
Looks good now with [CI build](https://devdiv.visualstudio.com/DevDiv/_build/results?buildId=5603163&view=results), I think Apex test timeouts are another issue.

**Review by erdembayar — APPROVED** [review]
👍 Thank you, couldn't wait this PR get merged.

File: test/NuGet.Clients.Tests/NuGet.CommandLine.Test/NuGetRestoreCommandTest.cs [hunk 1]
```diff
@@ -1913,6 +1922,8 @@ public async Task RestoreCommand_PreservesPackageFilesTimestamps()
                 var dllFileInfo = new FileInfo(dllPath);
                 Assert.True(File.Exists(dllFileInfo.FullName));
                 Assert.Equal(entryModifiedTime, dllFileInfo.LastWriteTime);
+
+                // TODO NK -
```
Comment (zivkan) [inline test/NuGet.Clients.Tests/NuGet.CommandLine.Test/NuGetRestoreCommandTest.cs]: todos should be resolved before merging.

Comment (nkolev92) [reply test/NuGet.Clients.Tests/NuGet.CommandLine.Test/NuGetRestoreCommandTest.cs]: oh nice catch. 

This was failing locally for me, but never failed on the CI. 

Removed.

**Pull request merged**